src/test/resources/edu/hm/hafner/coverage/parser/clover/clover.xml

```diff
@@ -0,0 +1,198 @@
+            <file name="File2.js" path="/home/jenkins/agent/workspace/dir/src/js/actions/File2.js">
+                <metrics statements="40" coveredstatements="33" conditionals="17" coveredconditionals="12" methods="115" coveredmethods="93"/>
+                <line num="21" count="19" type="stmt"/>
+                <line num="38" count="1" type="stmt"/>
+                <line num="51" count="163" type="stmt"/>
+                <line num="65" count="149" type="stmt"/>
+                <line num="79" count="1" type="stmt"/>
+                <line num="92" count="0" type="stmt"/>
+                <line num="105" count="1" type="stmt"/>
+                <line num="117" count="15" type="stmt"/>
+                <line num="127" count="0" type="stmt"/>
+                <line num="138" count="11" type="stmt"/>
+                <line num="151" count="25" type="stmt"/>
+                <line num="164" count="26" type="stmt"/>
+                <line num="176" count="24" type="stmt"/>
+                <line num="190" count="72" type="stmt"/>
+                <line num="204" count="0" type="stmt"/>
+                <line num="215" count="32" type="stmt"/>
+                <line num="228" count="32" type="stmt"/>
+                <line num="243" count="4" type="stmt"/>
+                <line num="257" count="24" type="stmt"/>
+                <line num="268" count="1" type="stmt"/>
+                <line num="287" count="10" type="stmt"/>
+                <line num="303" count="1" type="stmt"/>
+                <line num="317" count="5" type="stmt"/>
+                <line num="329" count="11" type="stmt"/>
+                <line num="339" count="1" type="stmt"/>
+                <line num="349" count="2" type="stmt"/>
+                <line num="359" count="4" type="stmt"/>
+                <line num="369" count="0" type="stmt"/>
+                <line num="380" count="44" type="stmt"/>
+                <line num="393" count="32" type="stmt"/>
+                <line num="405" count="2" type="stmt"/>
+                <line num="416" count="4" type="stmt"/>
+                <line num="429" count="1" type="stmt"/>
+                <line num="443" count="1" type="stmt"/>
+                <line num="456" count="8" type="stmt"/>
+                <line num="467" count="2" type="stmt"/>
+                <line num="480" count="2" type="stmt"/>
+                <line num="492" count="0" type="stmt"/>
+                <line num="503" count="0" type="stmt"/>
+                <line num="515" count="0" type="stmt"/>
+            </file>
+        </package>
+        <package name="components">
+            <metrics statements="46" coveredstatements="35" conditionals="12" coveredconditionals="8" methods="17" coveredmethods="7"/>
+            <file name="File3.jsx" path="/home/jenkins/agent/workspace/dir/src/js/components/File3.jsx">
+                <metrics statements="17" coveredstatements="11" conditionals="2" coveredconditionals="2" methods="8" coveredmethods="3"/>
```

Comment:
Why is there `conditionals="2" coveredconditionals="2"` but in line 61 we have no coverage of the false branch:

```
truecount="2" falsecount="0"
```

Reply:
Branches are currently not used so far (only the branch coverage). That means in the file rendering you see only green and red lines. If you want to support a visualization of the branches then you need to evaluate these values as well. Then it makes sense to see what `truecount="2" falsecount="0"` actually means.